### DISCUSSION TITLE
next/image is not honoring the `maxAge` set by upstream images

### DISCUSSION THREAD

**Discussion by RishikeshDarandale** [vercel/next.js#38327] [Help] [answered]
Hey team,
I see that documentation of next/image about the [caching behaviour](https://nextjs.org/docs/api-reference/next/image#caching-behavior) states that

> The expiration (or rather Max Age) of the optimized image is defined by either the minimumCacheTTL or the upstream image Cache-Control header, whichever is larger.

My upstream image has a header `cache-control: max-age=31536000`. When I use this image with next/image component, this `maxAge` value is not honoured at all and header is added like `Cache-Control: public, max-age=0, must-revalidate`.

When I looked at the [code](https://github.com/vercel/next.js/blob/12.1.7-canary.8/packages/next/server/image-optimizer.ts#L577), it correctly states that it will be always `Cache-Control: public, max-age=0, must-revalidate`.

This was got changed under [commit](https://github.com/vercel/next.js/commit/130f8645a2db18527142a7764eaa5e649183cc74), but the commit message does not talk more about why this was done.

Can you please help me understand is this a bug introduced in next/image?

//cc @styfle

**Comment by icyJoseph** [comment, marked as answer]
I think this has been fixed by https://github.com/vercel/next.js/pull/37625:

https://github.com/vercel/next.js/blame/42ee877e718753a80bec9f220aadf70c217a8745/packages/next/server/image-optimizer.ts#L576

Have you tried with version 12.2.0?

**Reply by RishikeshDarandale** [reply]
Thanks @icyJoseph , it works with `12.2.0`
